corebuild/integration/ILLink.Tasks/ILLink.CrossGen.targets

```diff
@@ -0,0 +1,231 @@
+  </PropertyGroup>
+
+  <!-- Rewrite inputs to ComputeFilesToPublish, so that crossgen'd
+       files get published instead of the pre-crossgen assemblies. -->
+  <Target Name="ComputeCrossGenFilesToPublish"
+          BeforeTargets="$(ComputeCrossGenFilesToPublishBefore)"
+          AfterTargets="$(ComputeCrossGenFilesToPublishAfter)"
+          Condition=" '$(CrossGenDuringPublish)' == 'true' "
+          DependsOnTargets="_CrossGenForPublish">
+
+    <!-- Rewrite ResolvedAssembliesToPublish, preserving metadata. -->
+    <ItemGroup>
+      <__CrossGenResolvedAssembliesToPublish Include="@(ResolvedAssembliesToPublish->'$(IntermediateOptimizedDir)/%(Filename)%(Extension)')" />
+      <_CrossGenResolvedAssembliesToPublish Include="@(__CrossGenResolvedAssembliesToPublish)" Condition="Exists('%(Identity)')" />
+      <!-- FilesToCrossGen doesn't contain System.Private.CoreLib, so
+           it will still be published. -->
+      <ResolvedAssembliesToPublish Remove="@(FilesToCrossGen)" />
+      <ResolvedAssembliesToPublish Include="@(_CrossGenResolvedAssembliesToPublish)" />
+    </ItemGroup>
+
+    <!-- Rewrite IntermediateAssembly, preserving metadata. -->
+    <ItemGroup>
+      <__CrossGenIntermediateAssembly Include="@(IntermediateAssembly->'$(IntermediateOptimizedDir)/%(Filename)%(Extension)')" />
+      <_CrossGenIntermediateAssembly Include="@(__CrossGenIntermediateAssembly)" Condition="Exists('%(Identity)')" />
+      <IntermediateAssembly Remove="@(FilesToCrossGen)" />
+      <IntermediateAssembly Include="@(_CrossGenIntermediateAssembly)" />
+    </ItemGroup>
+
+  </Target>
+
+  <!-- The target that runs crossgen on all input assemblies. We could
+       probably also reuse _RunCrossgen, but this gets its inputs from
+       ResolvedFileToPublish, so we would need to run crossgen after
+       ComputeFilesToPublish. -->
+  <Target Name="_CrossGenForPublish"
+          DependsOnTargets="PrepOptimizer;_ComputeCrossGenInputs">
+
+    <MakeDir Directories="$(IntermediateOptimizedDir)" />
+
+    <PropertyGroup>
+      <_CrossGenPlatformAssemblies>@(_CrossGenPlatformAssemblies)</_CrossGenPlatformAssemblies>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <CrossGenInvocations Include="$(MSBuildProjectFullPath)">
+        <Properties>
+          CrossGenExe=$(Crossgen);
+          CrossGenInput=%(FilesToCrossGen.FullPath);
+          CrossGenOutput=$(IntermediateOptimizedDir)/%(Filename)%(Extension);
+          CrossGenPlatformAssemblies=$(_CrossGenPlatformAssemblies)
+        </Properties>
+      </CrossGenInvocations>
+    </ItemGroup>
+
+    <MSBuild Projects="@(CrossGenInvocations)"
+             Targets="RunCrossGenForPublish" />
+
+  </Target>
+
+  <!-- Workarounds for SDK issues around the interdependency between
+       ComposeStore targets and CrossGen targets. These let us reuse
+       PrepOptimizer from the SDK to obtain a crossgen executable for
+       the target RID. -->
+
```

Comment:
Are these SDK issues expected to be fixed or will we get stuck with these workarounds?